gui/default/syncthing/core/syncthingController.js

```diff
@@ -651,13 +652,27 @@ angular.module('syncthing.core')
         };
 
         $scope.refreshFailed = function (page, perpage) {
-            var url = urlbase + '/folder/pullerrors?folder=' + encodeURIComponent($scope.failed.folder);
+            var url = urlbase + '/folder/' + $scope.failedType + 'errors'
+            url += '?folder=' + encodeURIComponent($scope.failed.folder);
             url += "&page=" + page + "&perpage=" + perpage;
             $http.get(url).success(function (data) {
                 $scope.failed = data;
             }).error($scope.emitHTTPError);
         };
 
+        $scope.totalFailed = function () {
+            if (!$scope.failed || !$scope.model[$scope.failed.folder]) {
+                return 0;
+            }
+            switch ($scope.failedType) {
+            case 'pull':
+                return $scope.model[$scope.failed.folder].pullErrors;
+            case 'scan':
+                return $scope.model[$scope.failed.folder].scanErrors;
+            }
+            return 0;
+        }
```

Comment:
This feels like a hack, just have two separate structs to store the stuff you want to store and separate functions for actions (they can be curried I guess)